portable/MemMang/heap_2.c

```diff
@@ -51,13 +52,16 @@
     #error This file must not be used if configSUPPORT_DYNAMIC_ALLOCATION is 0
 #endif
 
+#ifndef configHEAP_CLEAR_MEMORY_ON_FREE
+    #warning "configHEAP_CLEAR_MEMORY_ON_FREE is not defined. We recommend defining it to 1 in FreeRTOSConfig.h for better security."

```

Comment:
I have a concern that not all compilers understand a `#warning`. Maybe a `pragma message` would work better.

Reply:
The #warning is consistent with other usage in the codebase.  MSVC is the only compiler we support that I'm aware of not supporting it.  In this case however my preference is to remove the message warning message anyway - be that output by a #warning or #pragma.  The new configuration constant will be documented as per any other, and I think that is enough.